config.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package yarpc
+
+import (
+	"context"
+
+	"go.uber.org/yarpc/api/middleware"
+	"go.uber.org/yarpc/internal/observerware"
+	"go.uber.org/yarpc/internal/pally"
+
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// LoggingConfig describes how logging should be configured.
+type LoggingConfig struct {
+	// Supplies a logger for the dispatcher. By default, no logs are
+	// emitted.
+	Zap *zap.Logger
+	// If supplied, ExtractContext is used to log request-scoped
+	// information carried on the context (e.g., trace and span IDs).
+	ExtractContext func(context.Context) zapcore.Field
```

Comment:
Not wild about the naming here -`ContextExtractor` seems to answer the "What is this field?" question a bit better, e.g. "It's a ContextExtractor func".

Other ideas:

- `ExtractFromContext`
- `FromContext`
- Keep it the same

The first one in this list favors the "what does it do?", which is also nice.